libs/ngxtension/lazy-service/src/lazy-service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injector, Type, runInInjectionContext } from '@angular/core';
+import { assertInjector } from 'ngxtension/assert-injector';
+import { Observable, defer } from 'rxjs';
+
+/**
+ * Loads a service lazily. The service is loaded when the observable is subscribed to.
+ *
+ * @param loader A function that returns a promise of the service to load.
+ * @param injector The injector to use to load the service. If not provided, the current injector is used.
+ * @returns An observable of the service.
+ *
+ * @example
+ * ```ts
+ * const dataService$ = lazyService(() => import('./data-service').then((m) => m.MyService));
+ * or
+ * const dataService$ = lazyService(() => import('./data-service'));
+ * ```
+ */
+export function lazyService<T>(
+	loader: () => Promise<Type<T>> | Promise<{ default: Type<T> }>,
+	injector?: Injector
+): Observable<T> {
+	injector = assertInjector(lazyService, injector);
+
+	return runInInjectionContext(injector, () => {
```

Comment:
In which case is this necessary ?

Reply:
When we want to use it in ngOnInit

Reply:
But we're only calling `injector.get` ? Does that still requires to be in an injector context ? (if the injected class uses inject for example ?)

Reply:
You're right!